extensions/default/src/commandsModule.ts

```diff
@@ -0,0 +1,368 @@
+              viewportGridStore[storedHanging]
+            );
+          }
+        }
+        // Do this after successfully applying the update
+        stateSyncService.reduce(stateSyncReduce);
+        actions.toggleHpTools(hangingProtocolService.getActiveProtocol());
+        return true;
+      } catch (e) {
+        actions.toggleHpTools(hangingProtocolService.getActiveProtocol());
+        uiNotificationService.show({
+          title: 'Apply Hanging Protocol',
+          message: `The hanging protocol could not be applied due to ${e}`,
+          type: 'error',
+          duration: 3000,
+        });
+        return false;
+      }
+    },
+
+    toggleHangingProtocol: ({
+      protocolId,
+      stageIndex,
+    }: HangingProtocolParams): boolean => {
+      const {
+        protocol,
+        stage,
+        activeStudyUID,
+      } = hangingProtocolService.getActiveProtocol();
+      const { toggleHangingProtocol } = stateSyncService.getState();
+      const storedHanging = `${activeStudyUID}:${protocolId}:${stageIndex | 0}`;
+      if (
+        protocol.id === protocolId &&
+        (stageIndex === undefined || stageIndex === stage)
+      ) {
+        // Toggling off - restore to previous state
+        const previousState = toggleHangingProtocol[storedHanging] || {
+          protocolId: 'default',
+        };
+        return actions.setHangingProtocol(previousState);
+      } else {
+        stateSyncService.reduce({
+          toggleHangingProtocol: {
+            ...toggleHangingProtocol,
+            [storedHanging]: { protocolId: protocol.id, stageIndex: stage },
+          },
+        });
+        return actions.setHangingProtocol({ protocolId, stageIndex });
+      }
+    },
+
+    deltaStage: ({ direction }) => {
+      const state = viewportGridService.getState();
+      const { hangingProtocolId: protocolId, stageIdx } = state.hpInfo;
+      const { protocol } = hangingProtocolService.getActiveProtocol();
+      for (
+        let stageIndex = stageIdx + direction;
+        stageIndex >= 0 && stageIndex < protocol.stages.length;
+        stageIndex += direction
+      ) {
+        if (protocol.stages[stageIndex].enable !== 'disabled') {
+          return actions.setHangingProtocol({
+            protocolId,
+            stageIndex,
+          });
+        }
+      }
+      uiNotificationService.show({
+        title: 'Change Stage',
+        message: 'The hanging protocol has no more applicable stages',
+        type: 'error',
+        duration: 3000,
+      });
+    },
+
+    previousStage: () => {
+      const state = viewportGridService.getState();
+      const { reuseIdMap } = reuseCachedLayouts(state, stateSyncService);
+      // next stage in hanging protocols
+      hangingProtocolService.previousProtocolStage({ reuseIdMap });
+    },
+
+    /**
+     * Changes the viewport layout in terms of the MxN layout.
+     */
+    setViewportLayout: ({ numRows, numCols }) => {
```

Comment:
I think you mean `setViewerLayout` or even better, `setViewerGridLayout` (since it accepts rows and cols)